jolt-core/src/subprotocols/ra_virtual.rs

```diff
@@ -0,0 +1,204 @@
+use crate::{
+    field::JoltField,
+    poly::{
+        eq_poly::EqPolynomial,
+        multilinear_polynomial::{BindingOrder, MultilinearPolynomial, PolynomialBinding},
+    },
+    subprotocols::sumcheck::{BatchableSumcheckInstance, SumcheckInstanceProof},
+    utils::{math::Math, transcript::Transcript},
+};
+
+use crate::poly::multilinear_polynomial::PolynomialEvaluation;
+use rayon::iter::IndexedParallelIterator;
+use rayon::iter::{IntoParallelRefIterator, IntoParallelRefMutIterator, ParallelIterator};
+pub struct RAProof<F: JoltField, ProofTranscript: Transcript> {
+    pub ra_i_claims: Vec<F>,
+    pub sumcheck_proof: SumcheckInstanceProof<F, ProofTranscript>,
+}
+
+/// Virtual RA sumcheck for d-way chunked addresses
+pub struct RASumcheck<F: JoltField, const D: usize> {
```

Comment:
We probably want to have explicit separation here between the prover's sumcheck state and the verifier's sumcheck state (see [here](https://github.com/a16z/jolt/blob/feat/twist-shout-integration/jolt-core/src/subprotocols/shout.rs#L471-L475) for example) because this struct will be used by both prover and verifier.